MSAL/test/app/ios/MSALCacheItemDetailViewController.m

```diff
@@ -0,0 +1,121 @@
+{
+    UITextView *_cacheItemView;
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    [self setUpDetailView];
+    
+    // Do any additional setup after loading the view.
+}
+
+- (void)setUpDetailView
+{
+    if (self.cacheItem)
+    {
+        _cacheItemView = [[UITextView alloc] initWithFrame:self.view.bounds];
+        _cacheItemView.font = [UIFont systemFontOfSize:16.0f];
+        [self.view addSubview:_cacheItemView];
+        NSDictionary *jsonDict = [NSDictionary new];
+        
+        if ([self.cacheItem isKindOfClass:[MSIDBaseToken class]])
+        {
+            MSIDBaseToken *token = (MSIDBaseToken *)self.cacheItem;
+            MSIDCredentialCacheItem *item = token.tokenCacheItem;
+            jsonDict = item.jsonDictionary;
+        }
+        else if ([self.cacheItem isKindOfClass:[MSIDAppMetadataCacheItem class]])
+        {
+            MSIDAppMetadataCacheItem *appMetadata = (MSIDAppMetadataCacheItem *)self.cacheItem;
+            jsonDict = appMetadata.jsonDictionary;
+            
+        }
+        else if([self.cacheItem isKindOfClass:[MSIDAccount class]])
+        {
+            MSIDAccount *account = (MSIDAccount *)self.cacheItem;
+            jsonDict = account.jsonDictionary;
+        }
+        else if([self.cacheItem isKindOfClass:[MSALTestAppAsymmetricKey class]])
+        {
+            MSALTestAppAsymmetricKey *key = (MSALTestAppAsymmetricKey *)self.cacheItem;
+            NSMutableDictionary *keyDict = [NSMutableDictionary new];
+            [keyDict setObject:key.kid forKey:@"kid"];
+            [keyDict setObject:key.name forKey:@"label"];
+            jsonDict = keyDict;
+        }
+        else if([self.cacheItem isKindOfClass:[MSIDAccountMetadataCacheItem class]])
+        {
+            MSIDAccountMetadataCacheItem *accountMetadata = (MSIDAccountMetadataCacheItem *)self.cacheItem;
+            jsonDict = accountMetadata.jsonDictionary;
+        }
+        
+        NSError *error = nil;
+        NSData *jsonData = [NSJSONSerialization dataWithJSONObject:jsonDict options:NSJSONWritingPrettyPrinted error:&error];
+        if ([jsonData length] > 0)
+        {
+            NSLog(@"Successfully serialized the dictionary into data = %@", jsonData);
+            NSString *jsonString = [[NSString alloc] initWithData:jsonData
+                                                         encoding:NSUTF8StringEncoding];
+            NSLog(@"JSON String = %@", jsonString);
+            _cacheItemView.text = jsonString;
+        }
+        else
+        {
+            NSLog(@"An error happened = %@", error);
+        }
+    }
+}
+/*
+#pragma mark - Navigation
+
+// In a storyboard-based application, you will often want to do a little preparation before navigation
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    // Get the new view controller using [segue destinationViewController].
+    // Pass the selected object to the new view controller.
+}
+*/
```

Comment:
Remove the commented out code if not using it

Reply:
Addressed.